data_validation/secret_manger.py

```diff
@@ -0,0 +1,53 @@
+class SecretMangerBuilder:
+    def build(self, client_type):
+        """
+        :param client_type:
+        :return: secret manger instance currently support gcp secret manger
+        """
+        if client_type.lower() == "gcp":
+            return GCPSecretManger()
+        else:
+            raise Exception(f"{client_type} is not supported yet.")
+
+
+class GCPSecretManger:
+    """
+    GCPSecretManger: client to access secrets stored at GCP secret manger
+    """
+
+    def __init__(self):
+        # Import the Secret Manager client library.
+        from google.cloud import secretmanager
```

Comment:
Do we need to add google-cloud-secret-manager in our requirements?

Reply:
added the dependencies to setup.py